### DISCUSSION TITLE
Dividing a number by eg. 3 adds unexpected zeroes

### DISCUSSION THREAD

**Discussion by hellowodl** [PaulRBerg/prb-math#205] [Q&A] [answered]
Hey folks, I didn't want to instantaneously create an issue for this as it might be intended behavior and perhaps I'm stupid.

Today I ran into a strange issue where dividing a larger number by 3 creates numbers that have a whole bunch of zeroes smacked onto them.
```solidity
console2.log(sd(46000e18).div(sd(3)).unwrap());             // 15333333333333333333333333333333333333333
console2.log(sd(46000e18).mul(sd(1).div(sd(3))).unwrap());  // 15333333333333333318000
```

Is there something I'm not seeing/aware of or is this strange behavior?

**Comment by PaulRBerg** [comment, marked as answer]
Thanks for your question!

`sd(3)` is actually 3e-18 in normal numbers (recall that this is a fixed-point math library).

You perhaps wanted to divide by `sd(3e18)`.

**Reply by hellowodl** [reply]
Thanks a lot for the fast response and the reminder :)
